jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3579,6 +3631,11 @@ public Method getEnclosingMethod() throws SecurityException {
  * @see #isAnonymousClass()
  */
 public String getSimpleName() {
+	MetadataCache cache = getMetadataCache();
+	if (cache.cachedSimpleName != null) {
+		return cache.cachedSimpleName.get();
```

Comment:
what happens when the SoftReference gets cleared?

Reply:
oh, sorry, missed this, we need to check the referent existence as well, will fix that